src/test/ui/traits/trait-object-with-self-in-projection-output-good.rs

```diff
@@ -0,0 +1,23 @@
+// compile-pass
```

Comment:
Similarly a comment here would be good:

---

Regression test related to #56288. Check that a supertrait projection (of `Output`) that references `Self` can be ok if it is referencing a projection (of `Self::Target`, in this case). Note that we still require the user to manually specify both `Target` and `Output` for now.